acceptance/lib/puppet/acceptance/install_utils.rb

```diff
@@ -120,82 +120,99 @@ def install_repos_on(host, sha, repo_configs_dir)
+            platform_configs_dir
+          )
 
-          when /^(debian|ubuntu)-([^-]+)-(.+)$/
-            variant = $1
-            version = $2
-            arch = $3
+          list = fetch(
+            "http://builds.puppetlabs.lan/puppet/%s/repo_configs/deb/" % sha,
+            "pl-puppet-%s-%s.list" % [sha, version],
+            platform_configs_dir
+          )
 
-            deb = fetch(
-              "http://apt.puppetlabs.com/",
-              "puppetlabs-release-%s.deb" % version,
-              platform_configs_dir
-            )
+          repo_dir = fetch_remote_dir("http://builds.puppetlabs.lan/puppet/%s/repos/apt/%s" % [sha, version], platform_configs_dir)
 
-            list = fetch(
-              "http://builds.puppetlabs.lan/puppet/%s/repo_configs/deb/" % sha,
-              "pl-puppet-%s-%s.list" % [sha, version],
-              platform_configs_dir
-            )
+          on host, "rm -rf /root/*.list; rm -rf /root/*.deb; rm -rf /root/#{version}"
 
-            repo_dir = fetch_remote_dir("http://builds.puppetlabs.lan/puppet/%s/repos/apt/%s" % [sha, version], platform_configs_dir)
+          scp_to host, deb, '/root'
+          scp_to host, list, '/root'
+          scp_to host, repo_dir, '/root'
 
-            on host, "rm -rf /root/*.list; rm -rf /root/*.deb; rm -rf /root/#{version}"
+          on host, "mv /root/*.list /etc/apt/sources.list.d"
+          on host, "find /etc/apt/sources.list.d/ -name \"*.list\" -exec sed -i \"s/deb\\s\\+http:\\/\\/builds.puppetlabs.lan.*$/deb file:\\/\\/\\/root\\/#{version} #{version} main/\" {} \\;"
+          on host, "dpkg -i --force-all /root/*.deb"
+          on host, "apt-get update"
+        else
+          host.logger.notify("No repository installation step for #{platform} yet...")
+        end
+      end
 
-            scp_to host, deb, '/root'
-            scp_to host, list, '/root'
-            scp_to host, repo_dir, '/root'
+      def configure_gem_mirror(hosts)
+        hosts = [hosts] unless hosts.kind_of?(Array)
+        gem_source = ENV['GEM_SOURCE']
+        if gem_source.nil? || gem_source.empty?
+          default = ENV['OUTSIDE_PUPPETLABS'] ?
+            'https://rubygems.org' :
+            'http://rubygems.delivery.puppetlabs.net'
+          gem_source = default
+        end
 
-            on host, "mv /root/*.list /etc/apt/sources.list.d"
-            on host, "find /etc/apt/sources.list.d/ -name \"*.list\" -exec sed -i \"s/deb\\s\\+http:\\/\\/builds.puppetlabs.lan.*$/deb file:\\/\\/\\/root\\/#{version} #{version} main/\" {} \\;"
-            on host, "dpkg -i --force-all /root/*.deb"
-            on host, "apt-get update"
-          else
-            host.logger.notify("No repository installation step for #{platform} yet...")
+        hosts.each do |host|
+          on host, 'gem source -r https://rubygems.org/'
+          on host, 'gem source -r http://rubygems.org/'
+          on host, "gem source -a #{gem_source}"
```

Comment:
And maybe replace `-a` with `--add` (and `-r` with `--remove` if we keep those), so that the intent is more explicit to those (like me) who don't know the `gem source` command-line well :)